### PR TITLE
added load test sample code

### PR DESCRIPTION
**Pull request by poprahul**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+config:
+  ## Target will be your api endpoint base url
+  target: "<<APIGW ENDPOINT>>/<<STAGE>>"
+  phases:
+    - duration: 10
```
Comment (dmahapatro) [inline java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml:5]: Ideally, we should run it little bit higher duration (say, 60) with a higher arrival rate (say, 100)

File: java-test-samples/apigw-lambda-ddb/README.md [hunk 2]
```diff
@@ -91,6 +91,26 @@ private static final String STACK_NAME = "APIGW-Lambda-DDB-Sample";
...
+1. #### Install Artillery library
+``` npm install artillery@latest ```
+2. #### Install Faker library (_Optional. Only required to run sample load tests in this repo. Not a requirement to use Artillery library_)
+``` npm install faker ```
+3. #### Replace APIGW endpoint details in ```load-test-file.yaml``` and ```load-test-dynamic.yaml``` files
+4. #### ```load-test-file.yaml``` is the config file which shows how to use pre-defined test data from a csv file, whereas ```load-test-dynamic.yaml``` shows how you can have your own custom logic to generate random test data everytime.
+5. #### Run the load test using commands ````artillery run load-test-file.yml```` OR ``` artillery run load-test-dynamic.yml```
+
+For more information, please refer to the artillery [documentation](https://www.artillery.io/docs/)
+
```
Comment (dmahapatro) [inline java-test-samples/apigw-lambda-ddb/README.md:113]: How would one visualize the load rest result / benchmarking results? Refer [this section](https://github.com/aws-samples/serverless-java-frameworks-samples/tree/main/springboot#cloudwatch-logs-insights)

**Review by dmahapatro — CHANGES_REQUESTED** [review]

File: java-test-samples/apigw-lambda-ddb/README.md [hunk 3]
```diff
@@ -91,6 +91,26 @@ private static final String STACK_NAME = "APIGW-Lambda-DDB-Sample";
 Please make sure to replace the above with your stack name. This test submits an HTTP request to the API Gateway endpoint and verifies that the results are reflected in the DynamoDB database table.
 
 
+#### Load Test (using [Artillery](https://www.artillery.io/))
+Load tests should be executed in the cloud prior to any initial deployment to production environments. Load tests can be useful to discover performance bottlenecks and quota limits.  Load tests should simulate your application's expected peak load + 10% or more.
+
+There are several tools available for serverless developers to perform load testing. One of the most popular is `Artillery Community Edition`, which is an open-source tool for testing serverless APIs. You configure the number of requests per second and overall test duration, and it uses a headless Chromium browser to run its test flows.
+
+This project uses an open source performance testing tool called `Serverless Artillery`.
```
[the remark this answers is not in the document]
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:99]: The hyperlink is added for more info - "Load Test (using [Artillery](https://www.artillery.io/))"

File: java-test-samples/apigw-lambda-ddb/README.md [hunk 4]
```diff
@@ -91,6 +91,26 @@ private static final String STACK_NAME = "APIGW-Lambda-DDB-Sample";
 Please make sure to replace the above with your stack name. This test submits an HTTP request to the API Gateway endpoint and verifies that the results are reflected in the DynamoDB database table.
 
 
+#### Load Test (using [Artillery](https://www.artillery.io/))
+Load tests should be executed in the cloud prior to any initial deployment to production environments. Load tests can be useful to discover performance bottlenecks and quota limits.  Load tests should simulate your application's expected peak load + 10% or more.
+
+There are several tools available for serverless developers to perform load testing. One of the most popular is `Artillery Community Edition`, which is an open-source tool for testing serverless APIs. You configure the number of requests per second and overall test duration, and it uses a headless Chromium browser to run its test flows.
```
[the remark this answers is not in the document]
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:97]: There is an artillery pro version as well which provides cli extension and pro backend which enables scheduling & running tests from ECS/Fargate clusters. We are not using that in this repo.

File: java-test-samples/apigw-lambda-ddb/README.md [hunk 5]
```diff
@@ -91,6 +91,26 @@ private static final String STACK_NAME = "APIGW-Lambda-DDB-Sample";
 Please make sure to replace the above with your stack name. This test submits an HTTP request to the API Gateway endpoint and verifies that the results are reflected in the DynamoDB database table.
 
 
+#### Load Test (using [Artillery](https://www.artillery.io/))
+Load tests should be executed in the cloud prior to any initial deployment to production environments. Load tests can be useful to discover performance bottlenecks and quota limits.  Load tests should simulate your application's expected peak load + 10% or more.
```
[the remark this answers is not in the document]
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:95]: From Python readme file discussions in our team meetings.

File: java-test-samples/apigw-lambda-ddb/README.md [hunk 6]
```diff
@@ -91,6 +91,26 @@ private static final String STACK_NAME = "APIGW-Lambda-DDB-Sample";
...
+
+There are several tools available for serverless developers to perform load testing. One of the most popular is `Artillery Community Edition`, which is an open-source tool for testing serverless APIs. You configure the number of requests per second and overall test duration, and it uses a headless Chromium browser to run its test flows.
+
+This project uses an open source performance testing tool called `Serverless Artillery`. 
+
+The load test in this repo assumes the application stack provided in this sample is deployed on AWS as we need deployed api gateway endpoint details to run our load test against.
```
[the remark this answers is not in the document]
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:101]: Yes, only api gateway url is needed. And yes I can add sam build and sam deploy to be more clear.

File: java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml (hunk 1 above)
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml:5]: Tested with higher load but changed to lower values while doing more tests. Will change it back to bigger values.

File: java-test-samples/apigw-lambda-ddb/README.md (hunk 2 above)
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:113]: There is alway a report artillery generates. Will add that info in the READ me. And mentioned about cloudwatch metrics.

Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md:118]: done

File: java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml (hunk 1 above)
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/LoadTests/load-test-static-data.yml]: done

File: java-test-samples/apigw-lambda-ddb/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/README.md]: done

File: java-test-samples/apigw-lambda-ddb/LoadTests/load-test-dynamic-data.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+config:
+  ## Target will be your api endpoint base url
+  target: "<<APIGW ENDPOINT>>/<<STAGE>>"
```
Comment (dmahapatro) [inline java-test-samples/apigw-lambda-ddb/LoadTests/load-test-dynamic-data.yml:3]: SAM template output provides the `PaymentsEndpoint` including the stage name (`/dev`) and resource path (`/ticket`). That will mislead as to how users replace that url here. Wouldn't it be easier to [have a wrapper script](https://github.com/aws-samples/serverless-java-frameworks-samples/blob/main/springboot/load-test/run-load-test.sh) which gets the SAM template output and feed the url here instead? then update readme instructions to run the script?

Comment (poprahul) [reply java-test-samples/apigw-lambda-ddb/LoadTests/load-test-dynamic-data.yml]: Added script to fetch api endpoint from the cloudformation stack.

**Review by dmahapatro — APPROVED** [review]

**Pull request merged**